lib/api/v3/activities/activities_by_work_package_api.rb

```diff
@@ -0,0 +1,65 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'api/v3/activities/activity_representer'
+
+module API
+  module V3
+    module Activities
+      class ActivitiesByWorkPackageAPI < ::API::OpenProjectAPI
+        resource :activities do
+          helpers do
+            def save_work_package(work_package)
+              if work_package.save
+                journals = ::Journal::AggregatedJournal.aggregated_journals(
+                  journable: work_package)
+                Activities::ActivityRepresenter.new(journals.last, current_user: current_user)
```

Comment:
I don't think so. But:

* I think there is no quick way to do it better
* I did not introduce this problem, it was there before

This does not mean it shouldn't be solved, it just does not belong here and I hope that it is for now unimportant enough.